pkgs/by-name/wl/wl-shimeji/package.nix

```
@@ -0,0 +1,76 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  makeWrapper,
+  pkg-config,
+  which,
+  python3,
+  wayland,
+  wayland-scanner,
+  wayland-protocols,
+  libarchive,
+  uthash,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wl-shimeji";
+  version = "0.0.2";
+
+  src = fetchFromGitHub {
+    owner = "CluelessCatBurger";
+    repo = "wl_shimeji";
+    rev = "bf0150a313c3df271d249d3e14a17b48193f67e8";
+    hash = "sha256-ag3MHyyfo+2yZxPSBZgBsVZ4a6nLSv1KMUcYS34lIiM=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    wayland-protocols
+    wayland-scanner
+    libarchive
+    uthash
+    pkg-config
+    which
+    python3
+    makeWrapper
+  ];
+  buildInputs = [
+    wayland
+    libarchive
+  ];
+
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  installFlags = [
+    "PREFIX=$(out)"
+  ];
+
+  postInstall =
+    let
+      python = python3.withPackages (ps: with ps; [ pillow ]);
+    in
+    ''
+      cat <<EOF > $out/share/systemd/user/wl_shimeji.service
+      [Unit]
+      Description=wl_shimeji's overlay daemon
+
+      [Service]
+      Type=simple
+      ExecStart=$out/bin/shimeji-overlayd
+      EOF
+      wrapProgram $out/bin/shimejictl \
+        --set PYTHONPATH "${python.sitePackages}" \
+        --prefix PATH : ${lib.makeBinPath [ python ]}
+    '';
+
```

```suggestion
  postPatch = ''
    substituteInPlace systemd/wl_shimeji.service \
      --replace-fail "/usr/bin" "$out"
  '';
```
1. python wrapping is no longer necessary
2. Systemd service can be patch in a muuuuuch simpler way